fix/gsi/gsiparm.anl.sh

```diff
@@ -16,6 +16,7 @@ gsi_namelist="
    l4densvar=.false.,nhr_obsbin=3,
    use_gfs_nemsio=.false.,use_gfs_ncio=.true.,reset_bad_radbc=.true.,
    netcdf_diag=.false.,
+   l_obsprvdiag=.true.,
```

Comment:
It is okay to merge this PR first and then I will make this configurable through config.sh so that RRFS and RTMA can share the same gsiparm.anl.